<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <meta name="description" content="None">
        
        <link rel="canonical" href="https://sdi.genomics.hubs.altoslabs.com/rna_seq_investigation/">
        <link rel="shortcut icon" href="img/favicon.ico">
        <title>bulk RNA-seq Investigate</title>
        <link href="css/bootstrap.min.css" rel="stylesheet">
        <link href="css/font-awesome.min.css" rel="stylesheet">
        <link href="css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css">

        <script src="js/jquery-1.10.2.min.js" defer></script>
        <script src="js/bootstrap.min.js" defer></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script> 
    </head>

    <body class="homepage">
        <div class="navbar fixed-top navbar-expand-lg navbar-dark bg-primary">
            <div class="container">
                <a class="navbar-brand" href=".">bulk RNA-seq Investigate</a>
                <!-- Expander button -->
                <button type="button" class="navbar-toggler" data-toggle="collapse" data-target="#navbar-collapse">
                    <span class="navbar-toggler-icon"></span>
                </button>

                <!-- Expanded navigation -->
                <div id="navbar-collapse" class="navbar-collapse collapse">

                    <ul class="nav navbar-nav ml-auto">
                        <li class="nav-item">
                            <a href="#" class="nav-link" data-toggle="modal" data-target="#mkdocs_search_modal">
                                <i class="fa fa-search"></i> Search
                            </a>
                        </li>
                            <li class="nav-item">
                                <a href="https://github.com/wzheng0520/rna_seq_investigation/edit/master/docs/index.md" class="nav-link">Edit on rna_seq_investigation
                                    </a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
            <div class="row">
                    <div class="col-md-3"><div class="navbar-light navbar-expand-md bs-sidebar hidden-print affix" role="complementary">
    <div class="navbar-header">
        <button type="button" class="navbar-toggler collapsed" data-toggle="collapse" data-target="#toc-collapse" title="Table of Contents">
            <span class="fa fa-angle-down"></span>
        </button>
    </div>

    
    <div id="toc-collapse" class="navbar-collapse collapse card bg-secondary">
        <ul class="nav flex-column">
            
            <li class="nav-item" data-level="2"><a href="#1-bulk-rna-seq-issue-description" class="nav-link">1. Bulk RNA-seq Issue Description</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            
            <li class="nav-item" data-level="2"><a href="#cause-of-the-error" class="nav-link">Cause of the error</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            
            <li class="nav-item" data-level="2"><a href="#investigation-results" class="nav-link">Investigation Results</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            
            <li class="nav-item" data-level="2"><a href="#summary" class="nav-link">Summary</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            
            <li class="nav-item" data-level="2"><a href="#solution" class="nav-link">Solution</a>
              <ul class="nav flex-column">
              </ul>
            </li>
        </ul>
    </div>
</div></div>
                    <div class="col-md-9" role="main">

<h2 id="1-bulk-rna-seq-issue-description">1. Bulk RNA-seq Issue Description</h2>
<ul>
<li>The NFCORE_RNASEQ:RNASEQ:ALIGN_STAR:STAR_ALIGN step is notably slow when running bulk RNA-seq analysis on data from Genomics hubs AGR000166 and AGR000152. This slow operation causes the entire RNA-seq pipeline to fail, with errors reported as "Host EC2 (instance i-0486fb5f354ce15b1) terminated" or "Job attempt duration exceeded timeout.‚Äù</li>
</ul>
<table>
<thead>
<tr>
<th>Runs</th>
<th>Sequencer</th>
<th>Library Type</th>
<th>Reference Genome</th>
<th>Avg. FASTQ Size</th>
</tr>
</thead>
<tbody>
<tr>
<td>AGR000152</td>
<td>Novaseq 6000</td>
<td>mRNA-seq</td>
<td>GRCm39</td>
<td>3.5 GB</td>
</tr>
<tr>
<td>AGR000162</td>
<td>Novaseq 6000</td>
<td>mRNA-seq</td>
<td>GRCm39</td>
<td>1.6 GB</td>
</tr>
</tbody>
</table>
<h2 id="cause-of-the-error">Cause of the error</h2>
<ul>
<li>Alignement Error: <ul>
<li>It's possible that the issue stems from the samples not aligning properly with the reference genome.</li>
</ul>
</li>
<li>Reference Genome:<ul>
<li>It's pssible that the issue stems from the use inappropriate sample.</li>
</ul>
</li>
<li>Spike in:<ul>
<li>It's pssible that the issue stems from the merge ERCC spike in with reference genome.</li>
</ul>
</li>
<li>Total RNA-seq mix with mRNA-seq:<ul>
<li>It's pssible that the issue stems from the two mapped very slow runs are from the same data as the total RNA-seq mix.</li>
</ul>
</li>
<li>Pipeline updating<ul>
<li>It's pssible that the issue stems from pipline updated.</li>
</ul>
</li>
<li>Read Length<ul>
<li>It is possible that this problem stems from the fact that the mRNA needs to use a reads length of 101/101.</li>
</ul>
</li>
</ul>
<h2 id="investigation-results">Investigation Results</h2>
<ul>
<li>Research Design</li>
</ul>
<table>
<thead>
<tr>
<th>Alignement Error</th>
<th>Reference Genome</th>
<th>Spike in</th>
<th>Total RNA-seq mix with mRNA-seq</th>
<th>Pipeline updating</th>
<th>Read Length</th>
</tr>
</thead>
<tbody>
<tr>
<td>By downloading the trimmed fastq file and running it on personal EC2 by using the same command as nextflow</td>
<td>Alignment using GRCm38 as a reference genome</td>
<td>Do not add ERCC spike-in, use only GRCm39 as reference</td>
<td>Use the runs of the previous mix total RNA-seq (AGR000105) and mRNA-seq (AGR000108) to reanalyze</td>
<td>Use the former pipeline version (3.11.1) for analysis</td>
<td>Because normally, for mRNA, the reads length we would use is 101/101. but in this analysis, we are using 151/151, if I generate the reads length of 101/101 fastq file directly in the demux step, and then use the newly generated fastq for analysis</td>
</tr>
</tbody>
</table>
<ul>
<li>Research Results</li>
</ul>
<table>
<thead>
<tr>
<th>Alignement Error</th>
<th>Reference Genome</th>
<th>Spike in</th>
<th>Total RNA-seq mix with mRNA-seq</th>
<th>Pipeline updating</th>
<th>Read Length</th>
</tr>
</thead>
<tbody>
<tr>
<td>The downloaded fastq can run alignment on personal EC2 normally but it is also very slow</td>
<td>Despite the replacement of the reference genome, the alignment is still very slow</td>
<td>Despite not adding ERCC spike in, the comparison is still very slow</td>
<td>AGR000108(mRNAseq) can be analyzed normally</td>
<td>Despite using an older version of the rna-seq pipeline, the alignment is still very slow</td>
<td>After trimming the reads length of fastq to 101/101, RNA-seq can be analyzed normally. But once we also used 151/151 reads length to analyze novaseq mRNA data (AGR000177[GRCm39], AGR000108[GRCh38]), the previous results showed that they could be analyzed normally and very fast</td>
</tr>
</tbody>
</table>
<ul>
<li>Runs used for surveys</li>
</ul>
<table>
<thead>
<tr>
<th>Request ID</th>
<th>Read length</th>
<th>Reference</th>
<th>Spike in</th>
<th>RNAseq type</th>
<th>Run results</th>
<th>Sequencer</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>AGR000105</td>
<td>151/151</td>
<td>GRCm39</td>
<td>No</td>
<td>Total RNA</td>
<td>Success</td>
<td>Novaseq</td>
<td>Previous run have mixed total RNA-seq and mRNA-seq</td>
</tr>
<tr>
<td>AGR000108</td>
<td>151/151</td>
<td>GRCh38</td>
<td>No</td>
<td>mRNA</td>
<td>Success</td>
<td>Novaseq</td>
<td></td>
</tr>
<tr>
<td>AGR000166</td>
<td>151/151</td>
<td>GRCm39</td>
<td>Yes</td>
<td>mRNA</td>
<td>Failed</td>
<td>Novaseq</td>
<td>Failed run: Alignement error, reference genome, Spike in, and pipeline updating</td>
</tr>
<tr>
<td>AGR000177</td>
<td>100/100</td>
<td>GRCm39</td>
<td>No</td>
<td>mRNA</td>
<td>Success</td>
<td>Novaseq</td>
<td>Runs mixed with total RNA-seq but can alignmenet fast when the reads length is 151/151. This run is also the same batch with AGR000166</td>
</tr>
<tr>
<td>AGR000177</td>
<td>151/151</td>
<td>GRCm39</td>
<td>No</td>
<td>mRNA</td>
<td>Success</td>
<td>Novaseq</td>
<td></td>
</tr>
<tr>
<td>AGR000152</td>
<td>151/151</td>
<td>GRCm39</td>
<td>Yes</td>
<td>mRNA</td>
<td>Failed</td>
<td>Novaseq</td>
<td>Failed run: Alignement error, reference genome, Spike in, and pipeline updating</td>
</tr>
<tr>
<td>AGR000166</td>
<td>101/101</td>
<td>GRCm39</td>
<td>Yes</td>
<td>mRNA</td>
<td>Success</td>
<td>Novaseq</td>
<td>Trimmed the fastq into 101/101, to test the does read length influence the alignement reads</td>
</tr>
</tbody>
</table>
<ul>
<li>STAR command example: </li>
</ul>
<pre><code>STAR \
    --genomeDir star \
    --readFilesIn input1/&lt;sample_name_read1.fq.gz&gt; input2/&lt;sample_name_read2.fq.gz&gt; \
    --runThreadN 12 \
    --outFileNamePrefix sample_name. \
     \
    --sjdbGTFfile GRCm39_ERCC92.gtf \
    --outSAMattrRGline 'ID:sample_name'  'SM:sample_name'  \
    --quantMode TranscriptomeSAM --twopassMode Basic --outSAMtype BAM Unsorted --readFilesCommand zcat --runRNGseed 0 --outFilterMultimapNmax 20 --alignSJDBoverhangMin 1 --outSAMattributes NH HI AS NM MD --quantTranscriptomeBan Singleend --outSAMstrandField intronMotif
</code></pre>
<h2 id="summary">Summary</h2>
<ul>
<li>Before Trimmed (example: 17807)</li>
</ul>
<pre><code>                                 Started job on |   Jun 06 02:44:50
                             Started mapping on |   Jun 06 08:27:47
                                    Finished on |   Jun 06 16:28:35
       Mapping speed, Million of reads per hour |   3.02

                          Number of input reads |   24166036
                      Average input read length |   273
                                    UNIQUE READS:
                   Uniquely mapped reads number |   19877682
                        Uniquely mapped reads % |   82.25%
                          Average mapped length |   268.68
                       Number of splices: Total |   22378269
            Number of splices: Annotated (sjdb) |   22365606
                       Number of splices: GT/AG |   22197559
                       Number of splices: GC/AG |   147882
                       Number of splices: AT/AC |   17590
               Number of splices: Non-canonical |   15238
                      Mismatch rate per base, % |   0.20%
                         Deletion rate per base |   0.00%
                        Deletion average length |   1.68
                        Insertion rate per base |   0.00%
                       Insertion average length |   1.33
                             MULTI-MAPPING READS:
        Number of reads mapped to multiple loci |   2680963
             % of reads mapped to multiple loci |   11.09%
        Number of reads mapped to too many loci |   34149
             % of reads mapped to too many loci |   0.14%
                                  UNMAPPED READS:
  Number of reads unmapped: too many mismatches |   0
       % of reads unmapped: too many mismatches |   0.00%
            Number of reads unmapped: too short |   1549093
                 % of reads unmapped: too short |   6.41%
                Number of reads unmapped: other |   24149
                     % of reads unmapped: other |   0.10%
                                  CHIMERIC READS:
                       Number of chimeric reads |   0
                            % of chimeric reads |   0.00%

</code></pre>
<ul>
<li>After Trimmed (example: 17807)</li>
</ul>
<pre><code>Started job on |    Jun 06 11:22:11
                             Started mapping on |   Jun 06 11:39:03
                                    Finished on |   Jun 06 12:12:18
       Mapping speed, Million of reads per hour |   43.60

                          Number of input reads |   24159587
                      Average input read length |   198
                                    UNIQUE READS:
                   Uniquely mapped reads number |   19744219
                        Uniquely mapped reads % |   81.72%
                          Average mapped length |   196.93
                       Number of splices: Total |   15770903
            Number of splices: Annotated (sjdb) |   15765405
                       Number of splices: GT/AG |   15648274
                       Number of splices: GC/AG |   101550
                       Number of splices: AT/AC |   14109
               Number of splices: Non-canonical |   6970
                      Mismatch rate per base, % |   0.17%
                         Deletion rate per base |   0.00%
                        Deletion average length |   1.62
                        Insertion rate per base |   0.00%
                       Insertion average length |   1.31
                             MULTI-MAPPING READS:
        Number of reads mapped to multiple loci |   2784616
             % of reads mapped to multiple loci |   11.53%
        Number of reads mapped to too many loci |   41540
             % of reads mapped to too many loci |   0.17%
                                  UNMAPPED READS:
  Number of reads unmapped: too many mismatches |   0
       % of reads unmapped: too many mismatches |   0.00%
            Number of reads unmapped: too short |   1578930
                 % of reads unmapped: too short |   6.54%
                Number of reads unmapped: other |   10282
                     % of reads unmapped: other |   0.04%
                                  CHIMERIC READS:
                       Number of chimeric reads |   0
                            % of chimeric reads |   0.00%
</code></pre>
<ul>
<li>Read base pair compairsion</li>
</ul>
<table>
<thead>
<tr>
<th>Total Bases after mapped</th>
<th>Total Bases before mapped</th>
<th>unmapped bases</th>
</tr>
</thead>
<tbody>
<tr>
<td>3026331473</td>
<td>3654905019</td>
<td>628573546</td>
</tr>
<tr>
<td>2217107204</td>
<td>2444671569</td>
<td>227564366</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>Based on this investigation, we think it might cause by a locus (at the end of the reads) which has very small exons and appears to be highly expressed in the data.</strong></li>
</ul>
<h2 id="solution">Solution</h2>
<ul>
<li>running STAR with <code>--seedPerWindowNmax 30</code><ul>
<li>Through adding <code>--seePerWindowNmax</code> the speed increase by a factor  </li>
<li>Using <code>--seePerWindowNmax</code> will reduce the mapping sensitivity<ul>
<li><code>--seePerWindowNmax 30</code> increase speed by a factor of 10, but lose ~1%</li>
</ul>
</li>
</ul>
</li>
</ul></div>
            </div>
        </div>

        <footer class="col-md-12">
            <hr>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script>
            var base_url = ".",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
        <script src="js/base.js" defer></script>
        <script src="search/main.js" defer></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="searchModalLabel" aria-hidden="true">
    <div class="modal-dialog modal-lg">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="searchModalLabel">Search</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
                <p>From here you can search these documents. Enter your search terms below.</p>
                <form>
                    <div class="form-group">
                        <input type="search" class="form-control" placeholder="Search..." id="mkdocs-search-query" title="Type search term here">
                    </div>
                </form>
                <div id="mkdocs-search-results" data-no-results-text="No results found"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div><div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="keyboardModalLabel" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="keyboardModalLabel">Keyboard Shortcuts</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>

<!--
MkDocs version : 1.4.3
Build Date UTC : 2023-06-08 22:41:16.675288+00:00
-->
